dask_jobqueue/sge.py

```diff
@@ -16,6 +16,9 @@ class SGECluster(JobQueueCluster):
     ----------
     queue : str
         Destination queue for each worker job. Passed to `#$ -q` option.
+    memory: str
+        The amount of memory requested per worker job. Passed to `#$ -l` option. Can be
+        specified in lieu of `resource_spec`.
```

Comment:
Not sure you should override default docstring here. The addition you mademoiselle should be implicit, even if it is not the case currently.

Reply:
Ok! Will take this out.

Reply:
@guillaumeeb, actually, I think the default docstring should be overridden, so as to make clear what memory is being specified.

As I learned today, there are multiple memory specification possibilities for grid engine. Two that I currently know of are `m_mem_free` and `h_vmem`. In this PR, I am forcing the `memory` keyword argument to be passed into `m_mem_free`, and I think this should be made explicit for `SGECluster` users. 

As far as I can tell, `m_mem_free` is the most commonly used option for memory specification in job scripts here at work. `h_vem` is less commonly used. I expect that `m_mem_free` should be the correct default option for 95% of `SGECluster` use cases.

The Zen of Python states that "explicit is better than implicit", and I think it'd be better to state explicitly in the docstring that memory is passed to `#$ -l m_mem_free`, so that the end-user knows that if they need to specify `h_vmem` instead/on top the requested worker RAM, they still can do so with the `resource_spec` keyword argument as well.

Reply:
I'm just not sure of what will append to the consolidated docstring, as memory is already defined in `JobQueueCluster` class in core.py:
````
    memory: str
        Total amount of memory per job
````

You may end up with two different memory entries in the docstring. This is worth testing, or maybe look at docrep module to see if it provide some tools for this case.

Reply:
Understood. Thank you for pointing that out! I will ensure that the docs are clear and unambiguous on this pointer.